### PR TITLE
Save and restore top-level configuration settings

### PR DESCRIPTION
**Pull request by bmcbarron**
Persist the state across Editor executions:
* window bounds
* window maximize state
* currently open project/workspace

Add new "Close Workspace" menu option, in case someone wants to close the editor and not remember the currently open workspace.

As a sanity check, always clear Settings.OpenProject when resetting Settings.WorkspacePath. The current app state doesn't allow them to be set at the same time, but this makes it more explicit.

### CODE REVIEW

File: src/main/main.ts [hunk 1]
```diff
@@ -40,13 +63,19 @@ export default class EditorApp {
...
 	public static ConfigureSettings(filePathArgument: Undefinable<string>): void {
 		if (!filePathArgument) { return; }
-		Settings.OpenedFile = null;
-		Settings.WorkspacePath = null;
 
 		const extension = extname(filePathArgument).toLowerCase();
 		switch (extension) {
-			case ".editorproject": Settings.OpenedFile = filePathArgument; break;
-			case ".editorworkspace": Settings.WorkspacePath = filePathArgument; break;
+			case ".editorproject": {
```
Comment (julien-moreau) [inline src/main/main.ts]: Braces are not useful here, to keep homogenous code we should remove. I put braces only when needed

File: src/main/main.ts [hunk 2]
```diff
@@ -61,24 +90,33 @@ export default class EditorApp {
 		}
 
 		const platform = os.platform();
-		if (platform === "darwin") {
-			return argv[2];
+		let index =  (platform === "darwin") ? 2 : 1;
+		while (index < argv.length && argv[index].startsWith('--')) {
```
Comment (julien-moreau) [inline src/main/main.ts:93]: Nice

File: src/renderer/editor/components/main-toolbar.tsx [hunk 3]
```diff
@@ -87,6 +87,7 @@ export class MainToolbar extends React.Component<IToolbarProps, IToolbarState> {
             <Menu>
                 <MenuItem text="Open Workspace..." icon={<Icon src="workspace.svg" />} onClick={() => this._menuItemClicked("project:open-workspace")} />
                 <MenuItem text="Reveal WorkSpace In File Explorer" disabled={!WorkSpace.HasWorkspace()} icon="document-open" onClick={() => this._menuItemClicked("project:open-worspace-file-explorer")} />
+                <MenuItem text="Close Workspace" disabled={!WorkSpace.HasWorkspace()} icon="cross" onClick={() => this._menuItemClicked("project:close-workspace")} />
```
Comment (julien-moreau) [inline src/renderer/editor/components/main-toolbar.tsx]: For the icon please use "<Icon src="times.svg" />", this is the one I use everywhere. Do you confirm that "times.svg" is enough?

**Comment by bmcbarron** [comment]
I made the two changes you asked, but it looks like I broke the review history due to squashing all of my commits into one. Maybe that wasn't correct. Sorry! This should be ready to go.

**Comment by julien-moreau** [comment]
Hi @bmcbarron !
Just tested and it worked perfectly!

The only issue I found is, when the last loaded workspace doesn't exist anymore, it fails.
We should check using `await fs.pathExists(...)` before applying the setting
Once done, ready to merge :)

File: src/renderer/editor/components/main-toolbar.tsx (hunk 3 above)
Comment (julien-moreau) [reply src/renderer/editor/components/main-toolbar.tsx]: Thanks! I'm going to change the other icons where I use the @blueprintjs ones :)

**Comment by bmcbarron** [comment]
Done! I did commit --amend, so I needed to do force push again. Let me know if you want me to continue to do that in the future, or if squashing should only happen at the end of review to avoid messing up review diffs.

**Comment by julien-moreau** [comment]
Perfect !! You rox !!

**Review by julien-moreau — APPROVED** [review]

**Pull request merged**